.github/workflows/ci_liners.yaml

```diff
@@ -0,0 +1,103 @@
+name: ci_linters
+on:
+  pull_request:
+    types: [ opened, synchronize, reopened ]
+  push:
+
+
+env:
+  CARGO_TERM_COLOR: always
+  RUST_BACKTRACE: full
+  RUSTFLAGS: -D warnings
+
+jobs:
+  set-runs-on:
+    name: test-runs-on
+    runs-on: ubuntu-latest
+    outputs:
+      matrix: ${{ steps.set-matrix.outputs.matrix }}
+    steps:
+      - id: set-matrix
+        run: |
+          runs_on=''
+          nervosnetwork_actor_list='"janx", "doitian", "quake", "xxuejie", "zhangsoledad", "jjyr", "TheWaWaR", "driftluo", "keroro520", "yangby-cryptape"'
+          actor="${{github.actor}}"
+          if [[ "${{github.event.head_commit.message}}" == *"runs-on:"* ]]; then
+            runs_on=`echo "${{github.event.head_commit.message}}"|awk -F 'runs-on:' '{print $2}' | awk -F ']' '{print $1}'`
+            runs_on=$runs_on' ]'
+            echo "runs_on 1 is "$runs_on
+          elif [[ -n ${{env.CI_RUNS_ON}} ]]; then
+            runs_on='${{env.CI_RUNS_ON}}'
+            echo "runs_on 2 is "$runs_on
+          elif [[ ${{github.repository_owner}} != "nervosnetwork" ]] || [[ ${{github.repository_owner}} == "nervosnetwork" && ${{github.event_name}} == 'pull_request' && $nervosnetwork_actor_list != *$actor* ]]; then 
+            runs_on=" [ ubuntu-18.04,macos-10.15,windows-2019 ]"
+            echo "runs_on 3 is "$runs_on
+          elif
+             runs_on=" [ self-hosted-ci-ubuntu-20.04,macos-10.15,windows-2019 ] "
+          echo  "runs_on final is "$runs_on
+          echo "::set-output name=matrix::{\"os\":[ubuntu-18.04,macos-10.15]}"
+  get-runs-on:
+    name: get-runs-on
+    needs: set-runs-on
+    runs-on: ubuntu-latest
+    strategy:
+      matrix: ${{fromJSON(needs.test-runs-on.outputs.matrix)}} 
+    steps:
+    - name: echo test-runs-on out out
+      run: echo  ${{fromJSON(needs.test-runs-on.outputs.matrix)}}
+  ci_liners:
+    name: ci_liners
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-20.04,macos-10.15]
+      fail-fast: true
+      max-parallel: 10
+    env:
+      ImageOS:  ${{ matrix.os }}
+      BUILD_BUILDID: ${{ github.event.ref }}
+    steps: 
+      - uses: actions/checkout@v2
+      - name: Linters
+        run: |
+          echo "liners test"
```

Comment:
We must check ci workflow statuses here, otherwise `check_runs` will wait until all the platforms has completed.

Reply:
Something like:

```
your-fancy-script-wrapper echo "do the work"
```

In the script:

- Run given command like `sudo`, here run `echo "do the work"`.
- According the the sub process exit status, set the ci status.
- Exit with the sub process exit status

Pseudo code using shell

```
#!/usr/bin/env bash

set -e
set -u
[ -n "${DEBUG:-}" ] && set -x || true

set +e
"$@"
EXIT_STATUS="$?"
set -e

if [ "$EXIT_STATUS" = 0 ]; then
    echo "Check whether the ci succeeds"
else
    echo "Fail the ci"
fi
exit $EXIT_STATUS
```